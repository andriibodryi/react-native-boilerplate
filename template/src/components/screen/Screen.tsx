import * as React from 'react';

import { ScreenTypes } from './screen.types';
import { isFixed } from './screen.presets';
import { ScreenWithoutScrolling, ScreenWithScrolling } from './components';

export const Screen: React.FC<ScreenTypes> = ({ preset, ...props }) => {
  if (isFixed(preset)) {
    return <ScreenWithoutScrolling {...props} />;
  }
  return <ScreenWithScrolling {...props} />;
};

export default Screen;
