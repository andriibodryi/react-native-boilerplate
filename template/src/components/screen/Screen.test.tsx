import React from 'react';
import renderer from 'react-test-renderer';

import { Screen } from './Screen';

describe('Screen', () => {
  it('should render scrollable screen', () => {
    const tree = renderer.create(<Screen preset="scroll" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render fixed screen', () => {
    const tree = renderer.create(<Screen preset="fixed" />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
