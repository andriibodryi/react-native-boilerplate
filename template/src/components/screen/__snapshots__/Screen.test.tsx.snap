// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen should render fixed screen 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "height": "100%",
        },
        Object {
          "backgroundColor": "#FFFFFF",
        },
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "height": "100%",
          "justifyContent": "flex-start",
          "width": "100%",
        },
        Object {},
      ]
    }
  />
</View>
`;

exports[`Screen should render scrollable screen 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      },
      Object {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  Test child
</View>
`;
